### ISSUE TITLE
Encoding eventually fails on Half-Life

### ISSUE THREAD

**Issue by clementgallet** [clementgallet/libTAS#322]
```
[libTAS f:1655] Thread 12866 (main) Sending 3145728 bytes to ffmpeg
[rawvideo @ 0x5555557dfc80] PACKET SIZE: 129548, STRIDE: 168
[rawvideo @ 0x5555557dfc80] Invalid buffer size, packet size 129548 < expected frame_size 3145728
Error while decoding stream #0:0: Invalid argument
```

**Comment by vadosnaprimer** [comment]
Looks familiar http://tasvideos.org/forum/viewtopic.php?p=490172#490172

**Comment by clementgallet** [comment]
Oh yeah, I forgot you had that issue, I thought it was related to avutil link error.

Do you still have issues to dump movies? This is the first time that it is happening to me, but it happens consistently with that game (at different frame and different invalid buffer size).

Maybe some pipe error? But I'm not getting any error on the `fwrite()` call to the pipe. Or maybe the packet headers are being corrupted?

**Comment by vadosnaprimer** [comment]
I had it with Undertale recently on Lubuntu, and the author didn't have that error on Mint.

**Comment by clementgallet** [comment]
It looks like it's not the encode. I'm getting a crash eventually. Backtrace is:
```
(gdb) bt
#0  0x00000000 in  ()
#1  0x0000001d in  ()
#2  0x00000063 in  ()
#3  0x00000000 in  ()
```
So the main thread's stack is corrupted.

**Comment by clementgallet** [comment]
May be fixed by 2c1ed75.
